### DISCUSSION TITLE
Configure OrchardCore as Identity Server

### DISCUSSION THREAD

**Discussion by mhflam** [OrchardCMS/OrchardCore#15958] [Q&A]
Hi

I would like to use OrchardCore as an Identity / Resource Authorizarion Server and need help with setting it up.

Basically I have an internal web application where users can access secured resources.

A) my-web-app.com
B) my-orchard-server.com

When unauthenticated users access A) it will redirect them to the sign-on page on B)

Upon successful sign-on it will redirect users back to A) with access to the permitted resources 

The users, user roles, secured resources /permissions are managed and defined in A)

B) can somehow check which resources are allowed based on the users permissions returned from A)

Is there a guide or help document I can follow to achieve the above? 

I’m very new to Orchard Core and Identity Authentication/ Resource Authorisation so any help will be much appreciated.

TIA!

**Comment by MarGraz** [comment]
Hi, I think you can start installing two Orchard Core features (from the Configuration> Features menu in the Admin backoffice):

1. OpenID Authorization Server
2. OpenId Token Validation (this is required if you want to use bearer token authentication in the same tenant as the authorization server)

After that you need to configure the OpenID Server and choose the `flow` you want to use, set redirect URI, etc.

Check also the official [documentation](https://docs.orchardcore.net/en/latest/reference/modules/OpenId/).

Hope it helps.
